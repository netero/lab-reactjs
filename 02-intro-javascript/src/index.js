const getImage=async ()=>{
    const apiKey='urkEobvHn5kRdd8vderabjmWAHhYiEPC';
    const response=await fetch(`https://api.giphy.com/v1/gifs/random?api_key=${apiKey}`);
    const {data}=await response.json();
    return data.images.original.url;
};

getImage().then(url=>{
    const img=document.createElement('img');
    img.src=url;
    document.body.append(img);
});
