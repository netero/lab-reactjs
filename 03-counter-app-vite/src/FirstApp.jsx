import PropTypes from 'prop-types';

export const FirstApp = ( { title } ) => {
    return (
        <>
            <h1>{ title }</h1>
            <p>Subtitulo</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    title: 'No hay titulo'
}