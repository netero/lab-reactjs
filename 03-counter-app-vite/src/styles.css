html, body{
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    padding: 1rem;
}
button{
    padding: 0.3rem;
    font-size: 1.3rem;
    margin-right: 0.1rem;
}