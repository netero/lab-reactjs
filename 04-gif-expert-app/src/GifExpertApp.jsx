import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'Goku'])

    const onAddCategory = (category)=>{
        if(category.trim().length<=1)return;
        if(categories.includes(category)) return;
        setCategories([category, ...categories ]);
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory onAddCategory={onAddCategory}/>
            
            {categories.map(category => (
                <GifGrid 
                    key={category} 
                    category={category}/>
                )
            )}
        </>
    )
}