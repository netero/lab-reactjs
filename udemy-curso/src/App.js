import React, {Component} from 'react';
import logo from './logo.svg';
import './App.css';
import ConditionalSection from './sections/conditional';

function Hello(props){
  return <p>
  {props.title}
</p>
}


class Hello2 extends Component{
  render(){
    return <h3>{this.props.title}</h3>
  }
}

class Text extends Component{
  render(){
    return <div>
        <p>{this.props.text}</p>
        <p>{this.props.number}</p>
        <p>{this.props.boolean?"True":"False"}</p>
        <p>{this.props.arrayOfNumbers.map(this.props.multiply).join(', ')}</p>
        <p>{this.props.objectWithInfo.key}</p>
        {this.props.title}
      </div>
  }
}

class TextDefaultProp extends Component{
  render(){
    return <h1>{this.props.title}</h1>
  }
}

TextDefaultProp.defaultProps={
  title:"Esto es un tÃ­tulo por defecto."
}

class Contador extends Component{
  constructor(props){
    super(props)
    this.state={contador:props.contadorInicial}
    setInterval(()=>{
      this.setState({contador:this.state.contador+1})
    },1000)
  }
  render(){
    return <ContadorNumero numero={this.state.contador}/>
  }
}

class ContadorNumero extends Component{
  render(){
    return <span>{this.props.numero}</span>
  }
}

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <Hello title="Bienvenidos a este curso."/>
        <Hello2 title="Bienvenidos a este curso - Con clases."/>
        <Text
          arrayOfNumbers={[2,3,10]}
          boolean={true} 
          number={2} 
          objectWithInfo={{key:'value', key2:'otherValue'}}
          text="hola" 
          multiply={(n)=>n*3}
          title={<h1>Esto es un elemento de react que se pasa como prop</h1>}
          />
          <TextDefaultProp/>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Vamos a aprender React
        </a>
        <h3>Estoy usando react</h3>
        <Contador contadorInicial={55}/>
        <ConditionalSection/>
      </header>
    </div>
  );
}

export default App;
