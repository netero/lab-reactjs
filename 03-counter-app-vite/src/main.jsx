import React from "react";
import ReactDOM  from 'react-dom/client';
//import {FirstApp} from './FirstApp';
import {CounterApp} from './CounterApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={ 500 }/>
        {/*<FirstApp title="Title" subtitle="Subtitle"/>*/}
    </React.StrictMode>
)