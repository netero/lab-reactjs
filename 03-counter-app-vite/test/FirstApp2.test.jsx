import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en FirstApp", ()=>{
    const title ='Hola soty Rafael!!';
    const subtitle='Soy un subtitulo!!';
    test('DEbe hacer match con el snapshpt',()=>{
      const {container} = render(<FirstApp title={title}/>)  ;
      expect(container).toMatchSnapshot();
    });
    test('Debe mostrar el mensaje "Hola soy Rafael!!"',()=>{
      render(<FirstApp title={title}/>);
      expect(screen.getByText(title)).toBeTruthy();
    })
    test('Debe mostrar el titulo en un h1',()=>{
      render(<FirstApp title={title}/>);
      expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
    });

    test('DEbe mostrar el subtitulo enviado por props',()=>{
      render(<FirstApp title={title} subtitle={subtitle}/>);
      expect(screen.getByText(subtitle)).toBeTruthy();
    });
});