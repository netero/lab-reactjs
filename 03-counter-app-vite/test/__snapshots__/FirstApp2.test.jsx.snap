// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp DEbe hacer match con el snapshpt 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soty Rafael!!
  </h1>
  <p />
</div>
`;
